PowerPointLabs/PowerPointLabs/CropLab/CropToSlide.cs

```diff
@@ -0,0 +1,229 @@
+        {
+            try
+            {
+                if (shapeRange.Count < 1)
+                {
+                    ThrowErrorCode(ErrorCodeForSelectionCountZero);
+                }
+
+                if (!IsPictureForSelection(shapeRange))
+                {
+                    ThrowErrorCode(ErrorCodeForSelectionNonPicture);
+                }
+
+                return true;
+            }
+            catch (Exception e)
+            {
+                if (handleError)
+                {
+                    ProcessErrorMessage(e);
+                    return false;
+                }
+
+                throw;
+            }
+        }
+
+        private static void VerifyIsSelectionValid(PowerPoint.Selection selection)
+        {
+            if (selection.Type != PowerPoint.PpSelectionType.ppSelectionShapes)
+            {
+                ThrowErrorCode(ErrorCodeForSelectionCountZero);
+            }
+        }
+
+        private static bool IsPictureForSelection(PowerPoint.ShapeRange shapeRange)
+        {
+            return (from PowerPoint.Shape shape in shapeRange select shape).All(IsPicture);
+        }
+
+        private static bool IsPicture(PowerPoint.Shape shape)
+        {
+            return shape.Type == Office.MsoShapeType.msoPicture ||
+                   shape.Type == Office.MsoShapeType.msoLinkedPicture;
+        }
+
+        private static void ThrowErrorCode(int typeOfError)
+        {
+            throw new Exception(typeOfError.ToString(CultureInfo.InvariantCulture));
+        }
+
+        private static void IgnoreExceptionThrown() { }
+
+        public static string GetErrorMessageForErrorCode(string errorCode)
+        {
+            var errorCodeInteger = -1;
+            try
+            {
+                errorCodeInteger = Int32.Parse(errorCode);
+            }
+            catch
+            {
+                IgnoreExceptionThrown();
+            }
+            switch (errorCodeInteger)
+            {
+                case ErrorCodeForSelectionCountZero:
+                    return ErrorMessageForSelectionCountZero;
+                case ErrorCodeForSelectionNonPicture:
+                    return ErrorMessageForSelectionNonPicture;
+                default:
+                    return ErrorMessageForUndefined;
+            }
+        }
+
+        private static void ProcessErrorMessage(Exception e)
+        {
+            //This method prompts the error message to user. If it has an unrecognised error code,
+            //an alternative message window with erro trace stack pops up and prompts the user to
+            //send the trace stack to the developer team.
+            var errMessage = GetErrorMessageForErrorCode(e.Message);
+            if (!string.Equals(errMessage, ErrorMessageForUndefined, StringComparison.Ordinal))
+            {
+                MessageBox.Show(errMessage, MessageBoxTitle);
```

Comment:
have u tried to refactor this message box & dialog? using injection or other way.
it makes ur codes untestable in UT